gen/suites/actor_deletion/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/test-vectors/chaos"
+	. "github.com/filecoin-project/test-vectors/gen/builders"
+	. "github.com/filecoin-project/test-vectors/schema"
+)
+
+func main() {
+	g := NewGenerator()
+	defer g.Wait()
+
+	g.MessageVectorGroup("no_beneficiary",
+		&MessageVectorGenItem{
+			Metadata: &Metadata{
+				ID:      "delete-w-zero-balance",
+				Version: "v1",
+				Desc:    "actor with zero balance is deleted, does not require beneficiary",
+			},
+			Selector: Selector{"chaos_actor": "true"},
+			Func:     deleteActor,
+		},
+	)
+
+	g.MessageVectorGroup("beneficiary",
+		&MessageVectorGenItem{
+			Metadata: &Metadata{
+				ID:      "delete-w-balance-and-beneficiary",
+				Version: "v1",
+				Desc:    "actor with non-zero balance is deleted and sends funds to beneficiary",
+			},
+			Selector: Selector{"chaos_actor": "true"},
+			Func:     deleteActorWithBeneficiary(big.NewInt(50), address.Undef, exitcode.Ok),
+		},
+		// FIXME: this currently panics
+		// &MessageVectorGenItem{
+		// 	Metadata: &Metadata{
+		// 		ID:      "fail-delete-w-balance-and-unkown-beneficiary",
+		// 		Version: "v1",
+		// 		Desc:    "fails when actor with non-zero balance is deleted but beneficiary address is unknown",
+		// 	},
+		// 	Selector: Selector{"chaos_actor": "true"},
+		// 	Func:     deleteActorWithBeneficiary(big.NewInt(50), MustNewSECP256K1Addr("!👹*_👹!"), exitcode.SysErrorIllegalActor),
+		// },
+		&MessageVectorGenItem{
```

Comment:
On master, the generator type names and methods have been simplified! You'll probably need to make a few replaces here:

* `g.{MessageVectorGroup=>Group}()`
* `{MessageVectorGenItem=>VectorDef}`
* `{=>Message}Func`